be/src/exec/pipeline/hashjoin/spillable_hash_join_probe_operator.cpp

```diff
@@ -0,0 +1,465 @@
+    auto spill_readers = _join_builder->spiller()->get_partition_spill_readers(_processing_partitions);
+    _latch.reset(_processing_partitions.size());
+    auto query_ctx = state->query_ctx()->weak_from_this();
+    for (size_t i = 0; i < _processing_partitions.size(); ++i) {
+        std::shared_ptr<spill::SpillerReader> reader = std::move(spill_readers[i]);
+        auto task = [this, state, reader, i, query_ctx]() {
+            if (query_ctx.lock()) {
+                _update_status(_load_partition_build_side(state, reader, i));
+            }
+            _latch.count_down();
+        };
+        // TODO:
+        RETURN_IF_ERROR(_executor->submit(std::move(task)));
+    }
+    return Status::OK();
+}
+
+void SpillableHashJoinProbeOperator::_update_status(Status&& status) {
+    if (!status.ok()) {
+        std::lock_guard guard(_mutex);
+        _operator_status = std::move(status);
+    }
+}
+
+StatusOr<ChunkPtr> SpillableHashJoinProbeOperator::pull_chunk(RuntimeState* state) {
+    if (spill_strategy() == spill::SpillStrategy::NO_SPILL) {
+        return HashJoinProbeOperator::pull_chunk(state);
+    }
+
+    auto all_probe_partition_is_empty = [this]() {
+        for (auto& _prober : _probers) {
+            if (!_prober->probe_chunk_empty()) {
+                return false;
+            }
+        }
+        return true;
+    };
+
+    if (all_probe_partition_is_empty() && _staging_chunk != nullptr) {
+        // all probe partition is empty. push staging chunk firstly
+        RETURN_IF_ERROR(_push_probe_chunk(state, _staging_chunk));
+        _staging_chunk = nullptr;
+    }
+
+    if (_is_finishing) {
+#ifndef NDEBUG
+        auto partitioned_writer = down_cast<spill::PartitionedSpillerWriter*>(_probe_spiller->writer().get());
+        for (const auto& [level, partitions] : partitioned_writer->level_to_partitions()) {
+            auto writer = down_cast<spill::PartitionedSpillerWriter*>(_join_builder->spiller()->writer().get());
+            auto& build_partitions = writer->level_to_partitions().find(level)->second;
+            DCHECK_EQ(build_partitions.size(), partitions.size());
+            size_t build_rows = 0;
+            for (size_t i = 0; i < partitions.size(); ++i) {
+                build_rows += build_partitions[i]->num_rows;
+            }
+            DCHECK_EQ(build_rows, _join_builder->spiller()->spilled_append_rows());
+            // CHECK if left table is the same as right table
+            // for (size_t i = 0; i < partitions.size(); ++i) {
+            //     DCHECK_EQ(partitions[i]->num_rows, build_partitions[i]->num_rows);
+            // }
+        }
+#endif
+    }
+
+    if (_processing_partitions.empty()) {
+        _acquire_next_partitions();
+    }
+
+    if (!_all_loaded_partition_data_ready()) {
+        RETURN_IF_ERROR(_load_all_partition_build_side(state));
+    }
+
+    if (!_all_loaded_partition_data_ready()) {
```

Comment:
_load_all_partition_build_side will submit task to IO threads. Some partitions are in memory and it can be processed very fast. So here it is possible that the partition is already loaded successfully